### ISSUE TITLE
Question about trainning dataset.

### ISSUE THREAD

**Issue by cheng123tao** [microsoft/LLM2CLIP#43]
MODEL=EVA02-CLIP-L-14-336
PRETRAINED=eva_clip
python -m torch.distributed.launch --nproc_per_node=8 \
	--use_env training/main.py \
        --enable-deepspeed \
        --grad-checkpointing \
        --name="T_vitl336_Rcc12mR_Rcc3m_4ep" \
        --save-frequency 1  \
        --zeroshot-frequency 1 \
        --report-to="tensorboard, wandb" \
        --wandb-project-name="LLM2CLIP" \
        --wandb-notes="EVA02-CLIP-L-14-336" \
        --train-data-list "data/cc3m/{00..00287}.tar;data/cc12m/{00..01001}.tar" \
        --train-num-samples-list 2873538  10000225 \
        --eval-data-file=training/eval_datasets.yaml \
        --pretrained=${PRETRAINED} \
        --dataset-resampled \
        --precision "fp16" \
        --warmup 0 \
        --batch-size=512 \
        --eval-batch-size=1024 \
        --log-every-n-steps 50 \
        --epochs=20 \
        --lr=1e-5 \
        --visual-lr=1e-5 \
        --text-lr=1e-5 \
        --wd=0.05 \
        --visual-wd=0.05 \
        --text-wd=0.05 \
        --ld=1.0 \
        --text-ld=1.01 \
        --visual-ld=0.85 \
        --grad-clip-norm=5.0 \
        --smoothing=0. \
        --workers=8 \
        --model=${MODEL} \
        --seed 4096 \
        --gather-with-grad \
        --local-loss \
        --force-custom-clip \
        --optimizer="ap_adamw" \
        --zero-stage=1 \
        --dataset-type "webdataset" 

The data set in your training script is not the data after LLama embedding, but the original data. Did you use the embedded data for training? If not, what is the significance of processing the training data in the first step?

**Comment by raytrun** [comment]
In the first step, we process the training data to pre-extract text
embeddings, which accelerates model training and reduces VRAM usage for
improved efficiency.
Actually, the code has a hardcoded path for text embeddings. For reference,
please check line 522 in the data.py file from the repository:
https://github.com/microsoft/LLM2CLIP/blob/main/llm2clip/training/data.py#L522

Tao Cheng ***@***.***> 于2025年4月1日周二 14:22写道：
